### PR TITLE
Remove WPMediaPicker dependency

### PR DESCRIPTION
**Pull request by kean**
Removes WPMediaPicker dependency

To test:

- Smoke test Gutenberg: verify you can add images from Photos and Site Media
- Smoke test Site Media: verify that you can open video previews
- Smoke test Site Icon Picker: verify that picking from Site Media works

## Regression Notes
1. Potential unintended areas of impact: Media
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -643,41 +643,30 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
     }
 
     func gutenbergDidRequestMediaFromSiteMediaLibrary(filter: WPMediaType, allowMultipleSelection: Bool, with callback: @escaping MediaPickerDidPickMediaCallback) {
-        mediaPickerHelper.presentMediaPickerFullScreen(animated: true,
-                                                       filter: filter,
-                                                       dataSourceType: .mediaLibrary,
-                                                       allowMultipleSelection: allowMultipleSelection,
-                                                       callback: { [weak self] assets in
+        mediaPickerHelper.presentSiteMediaPickere(filter: filter, allowMultipleSelection: allowMultipleSelection) { [weak self] assets in
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]:
```suggestion
        mediaPickerHelper.presentSiteMediaPicker(filter: filter, allowMultipleSelection: allowMultipleSelection) { [weak self] assets in
```

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -3,6 +3,7 @@
 * [**] Add media fitlers to the Site Media screen [#22096]
 * [*] The "aspect ratio" mode on the Site Media screen is now also available on iPhone via the new title menu [#22096]
 * [**] Update the classic editor to use the new Photos and Site Media pickers [#22060]
+* [**] [internal] Remove WPMediaPicker dependency [#22103]
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt:6]: wdyt of combining this with the item below this one?

**Review by momo-ozawa — APPROVED** [review]
Works as described w a few comments

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: Fixed 😓

File: RELEASE-NOTES.txt (hunk 2 above)
Comment (kean) [reply RELEASE-NOTES.txt:6]: I changed the entry:

```
* [*] [internal] Rework Tenor (Free GIF) and Stock Photos (Free Photos) pickers [#22066, #22074]
```

The goal was to highlight the fact that Tenor and Stock photos changed. So it's more than just about removing WPMediaPicker, and it needs some testing.

**Pull request merged**